connector/src/it/scala/com/datastax/bdp/spark/search/SearchAnalyticsIntegrationSpec.scala

```diff
@@ -125,42 +126,25 @@ class SearchAnalyticsIntegrationSpec extends SparkCassandraITFlatSpecBase with D
     options: Map[String, String] = Map.empty,
     tableName: String = table): Unit = {
 
-    val fullOptions = options ++ Map("table" -> tableName, "keyspace" -> ks)
-    val optionString = fullOptions.map { case (key, value) => s"""$key "$value"""" }.mkString(",")
+    spark.conf.set(s"spark.sql.catalog.$registerName", classOf[CassandraCatalog].getCanonicalName)
+    options.foreach { case (key, value) => spark.conf.set(s"spark.sql.catalog.$registerName.$key", value) }
 
-    val createDDL =
-      s"""CREATE TABLE IF NOT EXISTS $registerName
-         |USING org.apache.spark.sql.cassandra
-         |OPTIONS (
-         | $optionString
-         |)""".stripMargin
-
-    spark.sql(createDDL)
   }
 
   implicit class DFFunctions(df: DataFrame) {
-
     /**
       * Attempt to walk back dependency tree for the RDD and find
       * the CassandraTableScanRDD. This should only work correctly if
       * we are in a series of 1 to 1 mappings back to the source.
       */
     def getUnderlyingCqlWhereClause(): CqlWhereClause = {
-      def getUnderlyingCqlWhereClause(rdd: RDD[_]): CqlWhereClause = {
-        if (rdd.dependencies.nonEmpty) {
-          getUnderlyingCqlWhereClause(rdd.dependencies.head.rdd)
-        } else {
-          rdd.asInstanceOf[CassandraTableScanRDD[_]].where
-        }
-      }
-
-      getUnderlyingCqlWhereClause(df.rdd)
+      getCassandraScan(df.queryExecution.sparkPlan).cqlQueryParts.whereClause
     }
   }
 
 
   "SearchAnalytics" should "be able to use solr_query" in skipIfNotDSE(conn) {
-    val df = spark.sql(s"""SELECT key,a,b,c FROM $table WHERE solr_query = '{"q": "*:*", "fq":["key:[500 TO *]", "a:1"]}' """)
+    val df = spark.sql(s"""SELECT key,a,b,c FROM $table.$ks.$table WHERE solr_query = '{"q": "*:*", "fq":["key:[500 TO *]", "a:1"]}' """)
```

Comment:
Noob question: what's with the $table.$ks.$table select here?  What's the leading $table for?

Reply:
This is weird because the way the test used to do this was to set options on a per table basis, registering each configuration as a specific table referring to similar underlying tables. To rewrite the tests with least amount of confusion I put each set of configuration into a catalog. So there is a catalog for which solr is turned on, there is a catalog for which solr is turned off, and so on. 

So here the leading identifier is for the catalog (which has the same name as the table)

Reply:
Ohhh, I see it now... registerTableWithOptions has all the fun.  Thx for the pointer. ;)